corehq/apps/export/forms.py

```diff
@@ -400,38 +411,190 @@ def get_multimedia_task_kwargs(self, export, download_id):
+                self._get_user_type_filter(mobile_user_and_group_slugs),
+            ])
+
+        form_filters += filter(None, [
+            self._get_users_filter(mobile_user_and_group_slugs),
+            self._get_locations_filter(mobile_user_and_group_slugs)
+        ])
+
+        form_filters = [OR(*form_filters)]
+        form_filters.append(self._get_datespan_filter())
+        return form_filters
+
+    def _get_group_filter(self, mobile_user_and_group_slugs):
+        """
+        :param mobile_user_and_group_slugs: ['g__e80c5e54ab552245457d2546d0cdbb03']
+        :return: GroupFormSubmittedByFilter filter which filters for users for selected groups
+        """
+        group_ids = self._get_group_ids(mobile_user_and_group_slugs)
+        if group_ids:
+            return GroupFormSubmittedByFilter(group_ids)
+
+    def get_user_ids_for_user_types(self, mobile, admin, unknown, demo, commtrack):
+        """
+        referenced from CaseListMixin to fetch user_ids for selected user type
+        :param mobile: if mobile users to be included
+        :param admin: if admin users to be included
+        :param unknown: if unknown users to be included
+        :param demo: if demo users to be included
+        :param commtrack: if commtrack users to be included
+        :return: user_ids for selected user types
+        """
+        from corehq.apps.es import filters, users as user_es
+        if not any([mobile, admin, unknown, demo]):
+            return []
+
+        user_filters = [filter_ for include, filter_ in [
+            (mobile, user_es.mobile_users()),
+            (admin, user_es.admin_users()),
+            (unknown, filters.OR(user_es.unknown_users(), user_es.web_users())),
+            (demo, user_es.demo_users()),
+        ] if include]
+
+        query = (user_es.UserES()
+                 .domain(self.domain_object.name)
+                 .OR(*user_filters)
+                 .show_inactive()
+                 .fields([]))
+        user_ids = query.run().doc_ids
+
+        if commtrack:
+            user_ids.append("commtrack-system")
+        if demo:
+            user_ids.append("demo_user_group_id")
+            user_ids.append("demo_user")
+        return user_ids
+
+    def _get_user_type_filter(self, mobile_user_and_group_slugs):
+        """
+        :param mobile_user_and_group_slugs: ['t__0', 't__1']
+        :return: FormSubmittedByFilter with user_ids for selected user types
+        """
+        user_types = self._get_es_user_types(mobile_user_and_group_slugs)
+        if user_types:
+            user_ids = self.get_user_ids_for_user_types(
+                mobile=HQUserType.REGISTERED in user_types,
+                admin=HQUserType.ADMIN in user_types,
+                unknown=HQUserType.UNKNOWN in user_types,
+                demo=HQUserType.DEMO_USER in user_types,
+                commtrack=False,
+            )
+            return FormSubmittedByFilter(user_ids)
```

Comment:
Absolutely.. Didn't notice that in my little 2 mobile worked test application. Thanks Mr. Hunt.